### PR TITLE
WIP: Feature/panel

### PR DESCRIPTION
**Pull request by jonnystening**
Ref: https://github.com/WestpacGEL/GEL/issues/39

### CODE REVIEW

File: components/panel/src/Panel.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	 */
+	children: PropTypes.node,
+};
+
+export const defaultProps = {
+	appearance: 'hero',
+	responsive: false,
+};
+
+Panel.propTypes = propTypes;
```
Comment (jaortiz) [inline components/panel/src/Panel.js]: Change to Panel.propTypes = { <props in here> }

File: components/panel/src/Panel.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+Panel.defaultProps = defaultProps;
```
Comment (jaortiz) [inline components/panel/src/Panel.js]: Same as above

File: components/panel/src/Panel.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import { jsx, useTheme } from '@westpac/core';
+import { paint } from './utils';
+
+// ==============================
+// Utils
+// ==============================
+
+export const PanelHeader = ({ appearance, responsive, ...props }) => {
+	const theme = useTheme();
+	const arrayValues = paint(theme.breakpoints);
```
[the remark this answers is not in the document]
Comment (jonnystening) [reply components/panel/src/Panel.js]: yeah I noticed this too. I used `arrayValues` as it's what Thinkmill used in Grid. I found it really difficult to understand what it was doing... except that it handled values provided as arrays.

There is a difference with this usage (which is using facepaint to turn values supplied as arrays into breakpoints), and the other usage I created for the Button component which handles arrays provided to props (ie. size and block props).

We really need to:

1. Come up with a good name for each that makes sense
1. Promote this functionality to Core?

File: components/panel/src/Panel.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+					: theme.panel.body.padding.default,
+			})}
+			{...props}
+		/>
+	);
+};
+
+export const PanelFooter = ({ appearance, responsive, ...props }) => {
+	const theme = useTheme();
+	const arrayValues = paint(theme.breakpoints);
```
[the remark this answers is not in the document]
Comment (jonnystening) [reply components/panel/src/Panel.js]: 🤔

File: components/panel/src/Panel.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+				},
+			})}
+			{...props}
+		/>
+	);
+};
+
+export const PanelBody = ({ appearance, responsive, ...props }) => {
+	const theme = useTheme();
+	const arrayValues = paint(theme.breakpoints);
```
[the remark this answers is not in the document]
Comment (jonnystening) [reply components/panel/src/Panel.js]: 🤔

File: components/panel/src/Panel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonnystening) [reply components/panel/src/Panel.js]: Moved this to Core and renamed `mq` 👍

**Pull request merged**